File: engine/model/src/main/scala/net/server/channel/packet/maker/MakerReagentPacket.scala
package net.server.channel.packet.maker

class MakerReagentPacket(private var _type: Int, private var _toCreate: Int, private var _isStimulant: Boolean, private var _reagentCount: Int, private var _reagentIds: Array[Int]) extends BaseMakerActionPacket(_type = _type, _toCreate = _toCreate) {
  def isStimulant: Boolean = _isStimulant

  def reagentCount: Int = _reagentCount

  def reagentIds: Array[Int] = _reagentIds
}

File: engine/engine-model/src/main/java/net/server/channel/packet/maker/MakerReagentPacket.java
package net.server.channel.packet.maker;

public class MakerReagentPacket extends BaseMakerActionPacket {
   private final boolean isStimulant;

   private final int reagentCount;

   private final int[] reagentIds;

   public MakerReagentPacket(int theType, int toCreate, boolean isStimulant, int reagentCount, int[] reagentIds) {
      super(theType, toCreate);
      this.isStimulant = isStimulant;
      this.reagentCount = reagentCount;
      this.reagentIds = reagentIds;
   }

   public boolean isStimulant() {
      return isStimulant;
   }

   public int reagentCount() {
      return reagentCount;
   }

   public int[] reagentIds() {
      return reagentIds;
   }
}
